### ISSUE TITLE
If there is a black box then how to update the parameters

### ISSUE THREAD

**Issue by Timeflies11** [mohammadpz/pytorch_forward_forward#16]
Hello, I would like to ask how such more parameters can be updated in a neural network with a black box, because it needs to use gradient descent to calculate the bias of w. If we do not know what's inside the black box, how to calculate the gradient descent.

**Comment by taomanwai** [comment]
I am confused about it too.

Maybe, there is an analog hardware component which could maximize its magnitude of output vector under a certain input through updating its internal states by **simple Physics rule(s)**.

Anyone with deep knowledge in FF/Physics can share?
